ui/src/controller/track_decider.ts

```diff
@@ -202,6 +202,93 @@ class TrackDecider {
     return 'Unknown';
   }
 
+  // Decorate an optional track description with additional contextual
+  // information. The base |description| itself may originate in the
+  // trace database or hard-coded in software.
+  static decorateTrackDescription(description: string | undefined,
+      args: Partial<{
+        upid: number|null,
+        processName: string|null,
+        numThreads: number|null,
+      }>): string | undefined {
+    const {upid, processName, numThreads} = args;
+
+    const hasProcessName = !!processName && upid != undefined;
```

Comment:
On another project (which has the lint rule configured that you mention) I was advised in the same situation to use this formulation because it also covers the case of `null`. i.e., `upid != undefined` is equivalent to `upid !== undefined && upid !== null`. It seems that ESLint is aware of this special case of the other otherwise deprecated `!=` operator and so doesn't flag it.

However, I am wary of the seeming inconsistency/magic involved and will prefer the more explicit, if more verbose, formulation. So I'll change it. Thanks!